### PR TITLE
Add assessment groups schema

### PR DESCRIPTION
**Pull request by SWhitlam**

### CODE REVIEW

**Review by Coupey — APPROVED** [review]

File: src/main/resources/db/migration/V1_5__pre_sentence_assessment.sql
Comment (adelaidaa) [inline src/main/resources/db/migration/V1_5__pre_sentence_assessment.sql:4]: mostly just this change

File: src/main/resources/db/migration/V1_7__rsr_only_assessment.sql
Comment (adelaidaa) [inline src/main/resources/db/migration/V1_7__rsr_only_assessment.sql:4]: and this

File: src/main/kotlin/uk/gov/justice/digital/assessments/services/QuestionService.kt
Comment (adelaidaa) [inline src/main/kotlin/uk/gov/justice/digital/assessments/services/QuestionService.kt:162]: @SWhitlam i have removed the coupling from then frontend endpoint to get groups by code and also uuid, this was necessary for the recursive load of groups so I have refactor that slightly so we just encapsulate getting the groups by uuid when needed in the service layer. I would like to merge this PR to test all of this is working as expected in the FE before i remove the top layer from the grouping table and we migrate the loading of the assessments to the new assessment schema layer.

**Review by adelaidaa — APPROVED** [review]
Approving after agreeing with @SWhitlam to merge and test this before we carry on with the replacement of groups by assessment schemas

**Pull request merged**
